File: vipclaw-admin/src/main/java/com/vipclaw/admin/dto/SysUserUpdateRequest.java
package com.vipclaw.admin.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;

/**
 * 用户更新请求 DTO
 *
 * @author vipamp
 * @since 2026-03-05
 */
@Data
@Schema(description = "用户更新请求对象")
public class SysUserUpdateRequest {

    /**
     * 用户 ID（更新时必须）
     */
    @Schema(description = "用户 ID", example = "1", requiredMode = Schema.RequiredMode.REQUIRED)
    private Long id;

    /**
     * 用户名（更新时不可修改）
     */
    @Schema(description = "用户名", example = "zhangsan", accessMode = Schema.AccessMode.READ_ONLY)
    private String username;

    /**
     * 密码
     */
    @Schema(description = "密码", example = "123456")
    @Size(min = 6, max = 100, message = "密码长度必须在 6-100 之间")
    private String password;

    /**
     * 昵称
     */
    @Schema(description = "昵称", example = "张三")
    @Size(max = 50, message = "昵称长度不能超过 50 个字符")
    private String nickname;

    /**
     * 邮箱
     */
    @Schema(description = "邮箱", example = "zhangsan@example.com")
    @Email(message = "邮箱格式不正确")
    private String email;

    /**
     * 手机号
     */
    @Schema(description = "手机号", example = "13800138000")
    @Pattern(regexp = "^1[3-9]\\d{9}$|^$", message = "手机号格式不正确")
    private String phone;

    /**
     * 性别 (0:女 1:男 2:未知)
     */
    @Schema(description = "性别 (0:女 1:男 2:未知)", example = "2")
    private Integer gender;

    /**
     * 头像 URL
     */
    @Schema(description = "头像 URL", example = "https://example.com/avatar.jpg")
    private String avatar;

    /**
     * 是否启用（0:禁用，1:启用）
     */
    @Schema(description = "是否启用（0:禁用，1:启用）", example = "1")
    private Integer status;
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/dto/SysUserUpdateRequest.kt
package com.vipamp.vipclaw.admin.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.*

/**
 * User update request object
 *
 * Defined according to database table structure:
 * - All fields are optional (partial update mechanism)
 * - Username is read-only and cannot be modified
 * - Null fields will not be updated
 */
@Schema(description = "User update request object")
data class SysUserUpdateRequest(
    @Schema(description = "User ID", example = "1")
    val id: Long? = null,

    @Schema(description = "Username", example = "zhangsan", accessMode = Schema.AccessMode.READ_ONLY)
    val username: String? = null,

    @Schema(description = "Nickname")
    @Size(max = 50, message = "Nickname length cannot exceed 50 characters")
    val nickname: String? = null,

    @Schema(description = "Email")
    @Email(message = "Invalid email format")
    val email: String? = null,

    @Schema(description = "Phone number")
    @Pattern(regexp = "^1[3-9]\\d{9}$|^$", message = "Invalid phone number format")
    val phone: String? = null,

    @Schema(description = "Gender (0:female 1:male 2:unknown)")
    val gender: Int? = null,

    @Schema(description = "Avatar URL", example = "https://example.com/avatar.jpg")
    val avatar: String? = null,

    @Schema(description = "Whether is administrator (0:no, 1:yes)")
    val isAdmin: Int? = null,
)
